File: JavaScript/9thOct-3.js
// Defining the Factorial Function First
function factorial(a_number) {
    if (a_number === 0){
        return 1;
    }
    return a_number * factorial(a_number - 1);
}

// Defining GCD Function
function gcdfact(a, b) {
    return factorial(Math.min(a, b));
    // By Using some Mathematical Command Sense xD
}


// Taking Input
const prompt = require("prompt-sync")();

const num_as_str = prompt("Please enter Your Numbers: ");

// Converting the String to Array
const num_args = num_as_str.split(/ +/g);

// Then to Number
const num_1 = Number(num_args[0]);
const num_2 = Number(num_args[1]);

// Calling the GCD Function
const res = gcdfact(num_1, num_2);
console.log(res);

File: C++/9thOct-3.cpp
#include <bits/stdc++.h>
using namespace std; 
  
int factorial(int x) 
{ 
    if (x <= 1) 
        return 1; 
    int res = 2; 
    for (int i = 3; i <= x; i++) 
        res = res * i; 
    return res; 
} 
  
int gcdOfFactorial(int m, int n) 
{ 
    return factorial(min(m, n)); 
} 
  
int main() 
{ 
    int m , n; 
    cin>>m>>n;
    cout << gcdOfFactorial(m, n); 
    return 0; 
} 
